In [88]:
#import cartopy.crs as ccrs
#import cartopy.feature as cfeature
import obspy
import matplotlib.patches as mpatches
import matplotlib.pyplot as plt
import numpy as np
from obspy import read
from obspy import UTCDateTime
from obspy.clients.fdsn import Client
from obspy.taup import plot_travel_times, TauPyModel
import pandas as pd 
from obspy.clients.fdsn.header import URL_MAPPINGS
#import pyproj

In [89]:

df_test = pd.read_csv("csvs/earthquake_metadata.csv")
#df_test = pd.read_csv("csvs/chunk5.csv")

df_test.set_index('trace_name',inplace= True)

C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3165: DtypeWarning: Columns (7,11,13,14,15,18,19,20,21,22,24,25,26,30,31) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [90]:
df_test.shape

(1265657, 34)

In [91]:
df_test['trace_start_time'] = pd.to_datetime(df_test['trace_start_time'],format="%Y-%m-%dT%H:%M:%S.%f")


In [92]:
df_test['station'] = df_test['network_code'] + '.' + df_test['receiver_code']

In [93]:
df_test['station']

trace_name
109C.TA_201510210555_NO     TA.109C
109C.TA_201511061450_NO     TA.109C
109C.TA_201511070220_NO     TA.109C
109C.TA_201511140515_NO     TA.109C
109C.TA_201512251850_NO     TA.109C
                             ...   
YHL.WY_20170621094042_EV     WY.YHL
YHL.WY_20170621122104_EV     WY.YHL
YHL.WY_20170621122339_EV     WY.YHL
YHL.WY_20170621133439_EV     WY.YHL
YHL.WY_20170621140610_EV     WY.YHL
Name: station, Length: 1265657, dtype: object

In [94]:
earliest_trace  = df_test[df_test['trace_start_time'] == df_test['trace_start_time'].min()]

In [95]:
df_test.head(1)

network_code receiver_code receiver_type  \
trace_name                                                         
109C.TA_201510210555_NO           TA          109C            HH   

                         receiver_latitude  receiver_longitude  \
trace_name                                                       
109C.TA_201510210555_NO            32.8889           -117.1051   

                         receiver_elevation_m  p_arrival_sample p_status  \
trace_name                                                                 
109C.TA_201510210555_NO                 150.0               NaN      NaN   

                         p_weight  p_travel_sec  ...  source_magnitude_author  \
trace_name                                       ...                            
109C.TA_201510210555_NO       NaN           NaN  ...                      NaN   

                        source_mechanism_strike_dip_rake  source_distance_deg  \
trace_name                                                                      
109C.TA_201510210555_NO                              NaN                  NaN   

                        source_distance_km back_azimuth_deg snr_db  \
trace_name                                                           
109C.TA_201510210555_NO                NaN              NaN    NaN   

                         coda_end_sample    trace_start_time trace_category  \
trace_name                                                                    
109C.TA_201510210555_NO              NaN 2015-10-21 05:55:00          noise   

                         station  
trace_name                        
109C.TA_201510210555_NO  TA.109C  

[1 rows x 35 columns]


for key in sorted(URL_MAPPINGS.keys()):
    client = Client(key)
#for key,client in clients_dict.items():
    try:
        station_test = client.get_stations(latitude=32.8889	, longitude=-117.1051)
        print(key)
    except:
        pass


In [96]:
earliest_trace

network_code receiver_code receiver_type  \
trace_name                                                          
CAL.NC_19840424221806_EV           NC           CAL            EL   

                          receiver_latitude  receiver_longitude  \
trace_name                                                        
CAL.NC_19840424221806_EV          37.450348         -121.800369   

                          receiver_elevation_m  p_arrival_sample p_status  \
trace_name                                                                  
CAL.NC_19840424221806_EV                 251.0             600.0   manual   

                          p_weight  p_travel_sec  ...  \
trace_name                                        ...   
CAL.NC_19840424221806_EV      0.51          4.64  ...   

                          source_magnitude_author  \
trace_name                                          
CAL.NC_19840424221806_EV                       NC   

                         source_mechanism_strike_dip_rake  \
trace_name                                                  
CAL.NC_19840424221806_EV                             None   

                          source_distance_deg source_distance_km  \
trace_name                                                         
CAL.NC_19840424221806_EV               0.2117              23.37   

                         back_azimuth_deg                              snr_db  \
trace_name                                                                      
CAL.NC_19840424221806_EV            147.0  [4.5        1.79999995 4.9000001 ]   

                          coda_end_sample        trace_start_time  \
trace_name                                                          
CAL.NC_19840424221806_EV        [[1900.]] 1984-04-24 22:18:07.960   

                            trace_category station  
trace_name                                          
CAL.NC_19840424221806_EV  earthquake_local  NC.CAL  

[1 rows x 35 columns]

In [97]:
#client = Client("IRIS")
client = Client("IRIS")
first_event = UTCDateTime(df_test['trace_start_time'].min().isoformat())
last_event = UTCDateTime(df_test['trace_start_time'].max().isoformat())



iris_events = client.get_events(starttime  = first_event,
endtime=last_event,
eventtype = "earthquake")
inventory = client.get_stations(latitude=37.34164, longitude=-121.64257	)
print(inventory)  

In [98]:
df_test_closest = df_test[df_test['source_distance_km'] == min(df_test['source_distance_km'])]

In [99]:
df_test_closest.dtypes

network_code                                object
receiver_code                               object
receiver_type                               object
receiver_latitude                          float64
receiver_longitude                         float64
receiver_elevation_m                       float64
p_arrival_sample                           float64
p_status                                    object
p_weight                                   float64
p_travel_sec                               float64
s_arrival_sample                           float64
s_status                                    object
s_weight                                   float64
source_id                                   object
source_origin_time                          object
source_origin_uncertainty_sec               object
source_latitude                            float64
source_longitude                           float64
source_error_sec                            object
source_gap_deg                 

In [100]:
from obspy import read_inventory, UTCDateTime
inv = read_inventory()

stations = client.get_stations(starttime  = first_event,
endtime=last_event,level='station')
#.get_contents()["stations"]
#["stations"]
#.get_contents()["stations"]

for key in sorted(URL_MAPPINGS.keys()):
    print(key)

#for s in stations.get_contents()["stations"]:
stations_per_provider = {}
providers = list(URL_MAPPINGS.keys())
faulty_providers = []
#providers.remove('BGR')
for key in providers:
    #client = Client(key)
    try:
        client = Client(key)
        stations_per_provider[key] = []
#client = Client("IRIS")
    
        stations = client.get_stations(starttime  = first_event,
        endtime=last_event,level='station')
        for s in stations:
            for e in s:
            #print()
                stations_per_provider[key].append(s.code + '.' + e.code)
    except:
        #print(key)
        faulty_providers.append(key)
        pass
        #raise ValueError
    #station_coords = {}
    #for s in stations:
    #    for e in s:
            #print()
    #        stations_per_provider[key].append(s.code + '.' + e.code)
            #station_coords[s.code + '.' + e.code] = (e.latitude,e.longitude)
            #print(s.code + '.' + e.code)
            #print(e.code)

    

stations_per_provider

        client = Client(key)
        stations_per_provider[key] = []
#client = Client("IRIS")
    
        stations = client.get_stations(starttime  = first_event,
        endtime=last_event,level='station')

Stations file creation

In [133]:
stations_df = pd.DataFrame(columns=['provider', 'network_name','station_name', 'latitude','longitude','station_opened','station_closed'])

In [102]:
stations_per_provider = {}
providers = list(URL_MAPPINGS.keys())
faulty_providers = []
#providers.remove('BGR')
#print(providers)
for key in providers:
    #client = Client(key)
    try:
        client = Client(key)
        stations_per_provider[key] = []
    #client = Client("IRIS")

        stations = client.get_stations(starttime  = first_event,
        endtime=last_event,level='station')
        for s in stations:
            for e in s:
                row_frame = pd.Series([key,s.code,e.code,e.latitude,e.longitude,e.creation_date,e.termination_date]).to_frame().T
                
                #row_frame = station_row.to_frame().T
                row_frame.columns = stations_df.columns

                stations_df = pd.concat([stations_df,row_frame],names = list(stations_df.columns), join = 'outer')
                #print(station_row)
                #stations_df = pd.concat([stations_df,station_row],ignore_index = False, axis = 0)
                #print(stations_df)
                
            #print()
                #stations_per_provider[key].append(s.code + '.' + e.code)
    except:
        #print(key)
        faulty_providers.append(key)
        pass

['BGR', 'ETH', 'EMSC', 'GEONET', 'GFZ', 'ICGC', 'INGV', 'IPGP', 'IRIS', 'ISC', 'KNMI', 'KOERI', 'LMU', 'NCEDC', 'NIEP', 'NOA', 'ODC', 'ORFEUS', 'RESIF', 'RASPISHAKE', 'SCEDC', 'TEXNET', 'UIB-NORSAR', 'USGS', 'USP']


C:\ProgramData\Anaconda3\lib\site-packages\obspy\io\stationxml\core.py:96: UserWarning: The StationXML file has version 1, ObsPy can read versions (1.0, 1.1). Proceed with caution.
  warnings.warn("The StationXML file has version %s, ObsPy can "


In [127]:
stations_df = stations_df.append(station_row,ignore_index=False)

TypeError: Can only append a Series if ignore_index=True or if the Series has a name

In [139]:

#to_frame().T
row_frame = station_row.to_frame().T
row_frame.columns = stations_df.columns

ee = pd.concat([stations_df,row_frame],names = list(stations_df.columns), join = 'outer')

#stations_df.append(station_row.to_frame().T,ignore_index=True,)

In [140]:
ee

provider network_name station_name latitude longitude  \
0      USP           II          ERM   42.015  143.1572   

                station_opened station_closed  
0  1990-05-21T00:00:00.000000Z           None

In [82]:
stations_df

Empty DataFrame
Columns: [provider, network_name, station_name, latitude, longitude, station_opened, station_closed]
Index: []

In [58]:
list(stations_df.columns)

['provider',
 'network_name',
 'station_name',
 'latitude',
 'longitude',
 'station_opened',
 'station_closed']

In [33]:
len(stations_df.columns)

7

In [34]:
station_row

0    0
1    1
2    2
3    3
4    4
5    5
6    6
7    7
dtype: int64

In [74]:
stations_df = pd.concat([stations_df, station_row.T], axis = 0, ignore_index=False, names = list(stations_df.columns))
#,names =  list(stations_df.columns))

In [75]:
stations_df

0
0               Empty DataFrame
Columns: [provider, network_na...
1               provider          0
network_name      1
statio...
provider                                                        0
network_name                                                    1
station_name                                                    2
latitude                                                        3
longitude                                                       4
station_opened                                                  5
station_closed                                                  6

In [53]:
stations_df = stations_df.append(station_row,ignore_index=True)

In [41]:
stations_df

provider network_name station_name latitude longitude station_opened  \
0      NaN          NaN          NaN      NaN       NaN            NaN   

  station_closed    0    1    2    3    4    5    6    7  
0            NaN  0.0  1.0  2.0  3.0  4.0  5.0  6.0  7.0

In [ ]:
stations_df

Empty DataFrame
Columns: [provider, network_name, station_name, latitude, longitude, station_opened, station_closed]
Index: []

In [ ]:
stations[0][0].elevation

185.0

In [ ]:
stations[0][0]

Station PTET (Porto Estrela, MT)
	Station Code: PTET
	Channel Count: None/None (Selected/Total)
	2016-05-04T00:00:00.000000Z - 2019-05-17T00:00:00.000000Z
	Access: open 
	Latitude: -15.38, Longitude: -57.16, Elevation: 185.0 m
	Available Channels:


In [ ]:
stations[0][0].code

'PTET'

In [ ]:
stations[0][0]

Station PTET (Porto Estrela, MT)
	Station Code: PTET
	Channel Count: None/None (Selected/Total)
	2016-05-04T00:00:00.000000Z - 2019-05-17T00:00:00.000000Z
	Access: open 
	Latitude: -15.38, Longitude: -57.16, Elevation: 185.0 m
	Available Channels:


In [ ]:
faulty_providers

['EMSC', 'GFZ', 'IRIS', 'ISC', 'USGS']

In [ ]:
faulty_providers

['EMSC', 'GFZ', 'IRIS', 'ISC', 'USGS']

In [ ]:
#for s in stations.get_contents()["stations"]:
#stations_per_provider = {}
#providers = list(URL_MAPPINGS.keys())
#faulty_providers = []
#providers.remove('BGR')
#faulty_providers.remove('EMSC')

#faulty_providers.remove('GFZ')
for key in faulty_providers:
    #client = Client(key)
    #try:
    print(key)
    client = Client(key)
    stations_per_provider[key] = []
#client = Client("IRIS")

    stations = client.get_stations(starttime  = first_event,
    endtime=last_event,level='station')
    for s in stations:
        for e in s:
        #print()
            try:
                stations_per_provider[key].append(s.code + '.' + e.code)
            except:
                pass
    #except:
    #    print(key)
        #faulty_providers.append(key)
        #pass
    #    raise ValueError
    #station_coords = {}
    #for s in stations:
    #    for e in s:
            #print()
    #        stations_per_provider[key].append(s.code + '.' + e.code)
            #station_coords[s.code + '.' + e.code] = (e.latitude,e.longitude)
            #print(s.code + '.' + e.code)
            #print(e.code)

    

IRIS


ValueError: value 336.68668 out of bounds (-90, 90)

In [ ]:
station_coords = {}
for s in stations:
    for e in s:
        #print()
        station_coords[s.code + '.' + e.code] = (e.latitude,e.longitude)
        #print(s.code + '.' + e.code)
        #print(e.code)


In [ ]:
df_iris_subset = df_test[df_test['station'].isin(station_coords.keys())]

In [ ]:
df_iris_subset

network_code receiver_code receiver_type  \
trace_name                                                            
AAK.II_20180115000700_NO             II           AAK            BH   
AAK.II_20180115001236_NO             II           AAK            BH   
AAK.II_20180115002306_NO             II           AAK            BH   
AAK.II_20180115003500_NO             II           AAK            BH   
AAK.II_20180115005518_NO             II           AAK            BH   
...                                 ...           ...           ...   
SNZO.IU_20161114015613_EV            IU          SNZO            BH   
SNZO.IU_20161115013448_EV            IU          SNZO            BH   
SNZO.IU_20161119204857_EV            IU          SNZO            BH   
SNZO.IU_20170204102921_EV            IU          SNZO            BH   
VAS01.ON_20141014200138_EV           ON         VAS01            HH   

                            receiver_latitude  receiver_longitude  \
trace_name                                                          
AAK.II_20180115000700_NO            42.637500           74.494200   
AAK.II_20180115001236_NO            42.637500           74.494200   
AAK.II_20180115002306_NO            42.637500           74.494200   
AAK.II_20180115003500_NO            42.637500           74.494200   
AAK.II_20180115005518_NO            42.637500           74.494200   
...                                       ...                 ...   
SNZO.IU_20161114015613_EV          -41.308700          174.704300   
SNZO.IU_20161115013448_EV          -41.308700          174.704300   
SNZO.IU_20161119204857_EV          -41.308700          174.704300   
SNZO.IU_20170204102921_EV          -41.308700          174.704300   
VAS01.ON_20141014200138_EV         -22.280081          -43.442624   

                            receiver_elevation_m  p_arrival_sample  \
trace_name                                                           
AAK.II_20180115000700_NO                  1633.1               NaN   
AAK.II_20180115001236_NO                  1633.1               NaN   
AAK.II_20180115002306_NO                  1633.1               NaN   
AAK.II_20180115003500_NO                  1633.1               NaN   
AAK.II_20180115005518_NO                  1633.1               NaN   
...                                          ...               ...   
SNZO.IU_20161114015613_EV                  120.0            1000.0   
SNZO.IU_20161115013448_EV                  120.0             700.0   
SNZO.IU_20161119204857_EV                  120.0             700.0   
SNZO.IU_20170204102921_EV                  120.0             800.0   
VAS01.ON_20141014200138_EV                 402.0             894.4   

                              p_status  p_weight  p_travel_sec  ...  \
trace_name                                                      ...   
AAK.II_20180115000700_NO           NaN       NaN           NaN  ...   
AAK.II_20180115001236_NO           NaN       NaN           NaN  ...   
AAK.II_20180115002306_NO           NaN       NaN           NaN  ...   
AAK.II_20180115003500_NO           NaN       NaN           NaN  ...   
AAK.II_20180115005518_NO           NaN       NaN           NaN  ...   
...                                ...       ...           ...  ...   
SNZO.IU_20161114015613_EV       manual      0.92         16.67  ...   
SNZO.IU_20161115013448_EV       manual      0.60         11.57  ...   
SNZO.IU_20161119204857_EV       manual      0.70         10.98  ...   
SNZO.IU_20170204102921_EV       manual      0.50         20.48  ...   
VAS01.ON_20141014200138_EV  autopicker      0.80          9.56  ...   

                            source_magnitude_author  \
trace_name                                            
AAK.II_20180115000700_NO                        NaN   
AAK.II_20180115001236_NO                        NaN   
AAK.II_20180115002306_NO                        NaN   
AAK.II_20180115003500_NO                        NaN   
AAK.II_20180115005518_NO                

In [ ]:
len(station_coords)

353

In [ ]:
stations

In [ ]:
len(stations['channels'])

2298

In [ ]:
coords

[]

In [ ]:
stations

['BL.APOB (FAZENDA SAO GERONIMO - APORE - GO)',
 'BL.APOB2 (FAZENDA SAO JOAO - APORE - GO)',
 'BL.AQDA (AQUIDAUANA - MS)',
 'BL.AQDB (Aquidauana, MS)',
 'BL.ARCA (Aracuai, MG)',
 'BL.AREB (AREADO - MG)',
 'BL.BAMB (BAMBUI - MG)',
 'BL.BB15B (FAZENDA CINTIA - BB - BEBEDOURO - SP)',
 'BL.BB16B (SITIO JOVAL - BB - BEBEDOURO - SP)',
 'BL.BB19B (Bebedouro 19, SP)',
 'BL.BB22B (Bebedouro 22, SP)',
 'BL.BEB (BELEM - PA)',
 'BL.BEB1 (SITIO STO ANTONIO - BEBEDOURO - SP)',
 'BL.BEB10 (FAZENDA SANTA ZULMIRA - BEBEDOURO - SP)',
 'BL.BEB11 (ESTANCIA DOURADA - ANDES - SP)',
 'BL.BEB12 (ESTANCIA MONIDENI - BEBEDOURO - SP)',
 'BL.BEB13 (FAZENDA RIO TURVO - BEBEDOURO - SP)',
 'BL.BEB14 (SITIO PALMEIRAS - BEBEDOURO - SP)',
 'BL.BEB15 (FAZENDA CINTIA  BEBEDOURO - SP)',
 'BL.BEB16 (SITIO JOVAL -  BEBEDOURO - SP)',
 'BL.BEB17 (SITIO SAO LEOPOLDO - BEBEDOURO - SP)',
 'BL.BEB18 (SITIO SAO PEDRO - BEBEDOURO - SP)',
 'BL.BEB19 (FAZENDA SANTA ZULMIRA - BEBEDOURO - SP)',
 'BL.BEB2 (FAZENDA SAO LUIZ -  BEBEDOURO 

In [ ]:
UTCDateTime()

AttributeError: 'str' object has no attribute 'isoformat'

In [ ]:
df_test_closest['network_code']

trace_name
KAN08.GS_20150408005359_EV    GS
Name: network_code, dtype: object

In [ ]:

client = Client("IRIS")
inventory = client.get_waveforms(network=df_test_closest['network_code'][0],
                                station=df_test_closest['receiver_code'][0],
                                starttime=UTCDateTime(df_test_closest['trace_start_time'][0].isoformat()),
                                endtime=UTCDateTime(df_test_closest['trace_start_time'][0].isoformat()) + 60,
                                location= "*",
                                channel =  "*",
                          attach_response=True)
                                #loc="*", 
                                #channel="*",
                                #level="response")  


FDSNException: Unknown Error (timeout): The read operation timed out

In [ ]:
earliest_trace['trace_start_time'][0].isoformat()

'1993-01-10T04:38:49.790000'

In [ ]:
clients_dict = {}
for key in sorted(URL_MAPPINGS.keys()):
    clients_dict[key] = Client(key)

In [ ]:
clients_dict.keys()

dict_keys(['BGR', 'EMSC', 'ETH', 'GEONET', 'GFZ', 'ICGC', 'INGV', 'IPGP', 'IRIS', 'ISC', 'KNMI', 'KOERI', 'LMU', 'NCEDC', 'NIEP', 'NOA', 'ODC', 'ORFEUS', 'RASPISHAKE', 'RESIF', 'SCEDC', 'TEXNET', 'UIB-NORSAR', 'USGS', 'USP'])

In [ ]:

for key in sorted(URL_MAPPINGS.keys()):
    client = Client(key)
#for key,client in clients_dict.items():
    try:
        inventory = client.get_waveforms(network=earliest_trace['network_code'][0],
                                    station=earliest_trace['receiver_code'][0],
                                    starttime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()),
                                    endtime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()) + 60,
                                    location= "*",
                                    channel =  "*",
                            attach_response=True)

        print(key)
    except:
        pass
        #print(key)

NCEDC


In [ ]:
inventory

21 Trace(s) in Stream:

BK.MHC..BHE | 1993-01-10T04:38:49.828000Z - 1993-01-10T04:39:49.778000Z | 20.0 Hz, 1200 samples
...
(19 other traces)
...
BK.MHC..VHZ | 1993-01-10T04:38:56.286000Z - 1993-01-10T04:39:46.286000Z | 0.1 Hz, 6 samples

[Use "print(Stream.__str__(extended=True))" to print all Traces]

In [ ]:

for key in sorted(URL_MAPPINGS.keys()):
    client = Client(key)
#for key,client in clients_dict.items():
    try:
        inventory = client.get_waveforms(network=earliest_trace['network_code'][0],
                                    station=earliest_trace['receiver_code'][0],
                                    starttime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()),
                                    endtime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()) + 60,
                                    location= "*",
                                    channel =  "*",
                            attach_response=True)

        print(key)
    except:
        pass
        #print(key)

NCEDC


In [ ]:
inventory

4 Trace(s) in Stream:
NC.CAL.02.EHZ | 1984-04-24T22:18:07.960000Z - 1984-04-24T22:19:07.950000Z | 100.0 Hz, 6000 samples
NC.CAL.02.ELE | 1984-04-24T22:18:07.960000Z - 1984-04-24T22:19:07.950000Z | 100.0 Hz, 6000 samples
NC.CAL.02.ELN | 1984-04-24T22:18:07.960000Z - 1984-04-24T22:19:07.950000Z | 100.0 Hz, 6000 samples
NC.CAL.02.ELZ | 1984-04-24T22:18:07.960000Z - 1984-04-24T22:19:07.950000Z | 100.0 Hz, 6000 samples

In [ ]:
earliest_trace

client = Client("IRIS")
inventory = client.get_waveforms(network=earliest_trace['network_code'][0],
                                station=earliest_trace['receiver_code'][0],
                                starttime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()),
                                endtime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()) + 60,
                                location= "*",
                                channel =  "*",
                          attach_response=True)

FDSNNoDataException: No data available for request.
Detailed response of server:



In [ ]:
wave_resampled = inventory.resample(8000)
wave_resampled = wave_resampled.normalize()
one_sample = np.array(wave_resampled[0])
#wave_norm_x =  one_sample/one_sample.std()
#sf.write('assets/new_sound_extraction_test_other.wav',wave_norm_x,8000)
sf.write('assets/new_sound_extraction_test_obspy_normalize.wav',one_sample,8000)


In [ ]:
import librosa
import soundfile as sf
#Extract the proper seismic instrument
#index: 0: north-south, 1: east-west, 2:vertical
def create_seismic_sound(x, index = 0):
    #x1 = x[:,index]
    x1=np.array(x)
    norm_x = x1 = x1/x1.std()
    #norm_x_resampled = librosa.resample(norm_x, 100, 8000)
    sf.write('assets/new_sound_extraction_test.wav',norm_x_resampled,8000)

    return norm_x_resampled

In [ ]:
waveform_normalize = inventory.normalize()

In [ ]:
np.array(waveform_normalize[0]).shape

(12001,)

In [ ]:
waveform_normalize_resampled = librosa.resample(np.array(waveform_normalize[0]), 200, 8000)
sf.write('assets/new_sound_extraction_test.wav',waveform_normalize_resampled,8000)

<ipython-input-65-f6bdc02d7f2b>:1: FutureWarning: Pass orig_sr=200, target_sr=8000 as keyword args. From version 0.10 passing these as positional arguments will result in an error
  waveform_normalize_resampled = librosa.resample(np.array(waveform_normalize[0]), 200, 8000)


In [ ]:
create_seismic_sound(inventory[0])

<ipython-input-49-b8532cfac02a>:9: FutureWarning: Pass orig_sr=100, target_sr=8000 as keyword args. From version 0.10 passing these as positional arguments will result in an error
  norm_x_resampled = librosa.resample(norm_x, 100, 8000)


array([-0.04705812, -0.04743128, -0.04779273, ...,  0.02808716,
        0.02323079,  0.01848785])

In [ ]:
df_test_closest

network_code receiver_code receiver_type  \
trace_name                                                            
KAN08.GS_20150408005359_EV           GS         KAN08            HH   

                            receiver_latitude  receiver_longitude  \
trace_name                                                          
KAN08.GS_20150408005359_EV           37.22672          -97.970947   

                            receiver_elevation_m  p_arrival_sample p_status  \
trace_name                                                                    
KAN08.GS_20150408005359_EV                 375.0             400.0   manual   

                            p_weight  p_travel_sec  ...  \
trace_name                                          ...   
KAN08.GS_20150408005359_EV      0.63          0.78  ...   

                            source_magnitude_type source_magnitude_author  \
trace_name                                                                  
KAN08.GS_20150408005359_EV                     ml                    ISMP   

                            source_mechanism_strike_dip_rake  \
trace_name                                                     
KAN08.GS_20150408005359_EV                              None   

                           source_distance_deg source_distance_km  \
trace_name                                                          
KAN08.GS_20150408005359_EV              0.0004               0.04   

                           back_azimuth_deg  \
trace_name                                    
KAN08.GS_20150408005359_EV            166.8   

                                                           snr_db  \
trace_name                                                          
KAN08.GS_20150408005359_EV  [54.40000153 51.5        48.40000153]   

                            coda_end_sample        trace_start_time  \
trace_name                                                            
KAN08.GS_20150408005359_EV        [[1052.]] 2015-04-08 00:54:00.800   

                              trace_category  
trace_name                                    
KAN08.GS_20150408005359_EV  earthquake_local  

[1 rows x 34 columns]